src/Microsoft.AspNet.Identity/IdentityLogger.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Identity
+{
+    public class IdentityLogger
+    {
+        public ILogger Logger { get; set; }
+
+        public virtual TResult LogResult<TResult>(TResult result, Func<TResult, LogLevel> getLevel,
+            Func<string> messageAccessor)
+        {
+            var logLevel = getLevel(result);
+
+            // Check if log level is enabled before creating the message.
+            if (Logger.IsEnabled(logLevel))
+            {
+                Logger.Log(logLevel, 0, messageAccessor(), null, (msg, exp) => (string)msg);
+            }
+
+            return result;
+        }
+
+        public virtual SignInResult LogSignInResult(SignInResult result, [CallerMemberName]string methodName = null)
```

Comment:
Wanted to differentiate what is being logged since it can be a either IdentityResult or SignInResult. I can change that